src/Persistence/SiteLinkItemPageUpdater.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace ProfessionalWiki\WikibaseFacetedSearch\Persistence;
+
+use MediaWiki\Page\WikiPageFactory;
+use MediaWiki\Title\Title;
+use MediaWiki\User\UserIdentity;
+use ProfessionalWiki\WikibaseFacetedSearch\Application\ItemPageUpdater;
+use Wikibase\DataModel\Entity\Item;
+use WikiPage;
+
+class SiteLinkItemPageUpdater implements ItemPageUpdater {
+
+	public function __construct(
+		private readonly string $linkTargetSitelinkSiteId,
+		private readonly WikiPageFactory $pageFactory
+	) {
+	}
+
+	public function updatePage( Item $item, UserIdentity $user ): void {
+		if ( $item->hasLinkToSite( $this->linkTargetSitelinkSiteId ) ) {
+			$this->getSiteLinkedPage( $item )
+				->newPageUpdater( $user )
+				->updateRevision();
+		}
+	}
+
+	private function getSiteLinkedPage( Item $item ): WikiPage {
+		return $this->pageFactory->newFromTitle(
+			Title::newFromText( $item->getSiteLink( $this->linkTargetSitelinkSiteId )->getPageName() )
```

Comment:
`Title::newFromText` can return `null`, and this is not purely theoretical, as there is no guarantee that the value in the sitelink represents a valid page name on the current wiki.